docs/src/main/sphinx/admin/properties.rst

```diff
@@ -21,3 +21,80 @@ may be used to tune Trino or alter its behavior when required.
     Logging <properties-logging>
     Web UI <properties-web-interface>
     Regular expression function <properties-regexp-function>
+
+Property value types
+--------------------
+
+Configuration properties in Trino take values of different types, some standard
+and some non-standard. Supported values may be further limited on a per-property
+basis.
+
+.. _prop-type-boolean:
+
+``boolean``
+^^^^^^^^^^^
+
+Properties of a ``boolean`` type support two values, ``true`` or ``false``
+in lowercase.
+
+.. _prop-type-data-size:
+
+``data size``
+^^^^^^^^^^^^^
+
+Properties of a ``data size`` type support values describing an amount of
+data, measured in byte-based units. For example, the value ``6GB`` describes
+six gigabytes.
+
+The ``data size`` type supports the following units:
+
+* ``B``: Bytes
+* ``kB``: Kilobytes
+* ``MB``: Megabytes
+* ``GB``: Gigabytes
+* ``TB``: Terabytes
+* ``PB``: Petabytes
+
+Some ``data size`` properties also support ``unlimited`` as a value.
+
+.. _prop-type-double:
+
+``double``
+^^^^^^^^^^
+
+Properties of a ``double`` type support numerical values, including decimals.
+
+.. _prop-type-duration:
+
+``duration``
+^^^^^^^^^^^^
+
+Properties of a ``duration`` type support values describing an amount of time.
+For example, the value ``7m`` describes seven minutes.
+
+The ``duration`` type supports the following units:
+
+* ``ns``: Nanoseconds
+* ``us``: Microseconds
+* ``ms``: milliseconds
+* ``s``: Seconds
+* ``m``: Minutes
+* ``h``: Hours
+* ``d``: Days
+
+.. _prop-type-integer:
+
+``integer``
+^^^^^^^^^^^
+
+Properties of an ``integer`` type support whole numerical values.
+
+.. _prop-type-string:
+
+``string``
+^^^^^^^^^^
+
+Properties of a ``string`` type support values consisting of a sequence of
```

Comment:
what about lenght restrictions or character sets?

Reply:
Properties of a ``string`` type are actually very rare. 
Do you have examples of free-form string-valued properties?

Reply:
Many properties are actually "enums".
I think it's so predominant we should introduce a separate type for them and leave "string" for really free-form string inputs.

Example of string property: `hive.temporary-staging-directory-path`
Example of enum property: `hive.storage-format` (default hive storage format, one of io.trino.plugin.hive.HiveStorageFormat enum)

Or, we continue to call all of them "string", and let users figure out. I am fine with that, but then we shouldn't make "string" sound like "free-form", as "values consisting of a sequence of characters" sounds to me

Reply:
We can explain it more but cant really easily change the docs to enum since string is used in a lot of places in the actual user facing docs. Just look at the diff for this PR where the text is introduced. Sure, they are often enums and we should explain that but they can also be freeform text.  We should explain that and have the valid values documented where there are enums.

Reply:
Without taking action on strings vs. enums, I reworded this ``string`` section a bit to try and not imply that it's free-form.